CMakeLists.txt

```diff
@@ -7,6 +7,11 @@ if(CCACHE_PROGRAM)
   set_property(GLOBAL PROPERTY RULE_LAUNCH_COMPILE "${CCACHE_PROGRAM}")
 endif()
 
+# Raise an error if attempting to compile on macOS older than 10.15
+if (APPLE)
+  set(CMAKE_OSX_DEPLOYMENT_TARGET "10.15" CACHE STRING "OSX deployment target")
+endif()
```

Comment:
Don't put this in a CMakeLists.txt file. Use the `MACOSX_DEPLOYMENT_TARGET` and / or pass it as `cmake -DCMAKE_OSX_DEPLOYMENT_TARGET:STRING=XX.XX` 

This is what we do in our workflows: 

https://github.com/NREL/EnergyPlus/blob/a844dbb1ab3424f6808d460cf638acb04d455ce1/.github/workflows/mac_release.yml#L68-L78

Reply:
Understood, I will revert this and rather update the build page on the Wiki since it mentions macOS 10.9 and higher: https://github.com/NREL/EnergyPlus/wiki/Building-EnergyPlus#mac

Reply:
I have updated the wiki to put 10.15 (FYI, we had to bump to 10.15 when we started using std::filesystem)

Reply:
I have removed forcing `CMAKE_OSX_DEPLOYMENT_TARGET` as requested, but instead added a fatal message if the value of this variable is set below the minimum supported version, because I find it nicer to warn upfront for unsupported configuration. Let me know what you think!